lib/remove-empty.js

```diff
@@ -0,0 +1,27 @@
+var Plugin = require('broccoli-plugin');
+var walkSync = require('walk-sync');
+var fs = require('fs');
+var path = require('path');
+
+module.exports = RemoveFiles;
+
+RemoveFiles.prototype = Object.create(Plugin.prototype);
+RemoveFiles.prototype.constructor = RemoveFiles;
+function RemoveFiles(inputNode, options) {
+  options = options || {};
+  Plugin.call(this, [inputNode], {
+    annotation: options.annotation
+  });
+}
+
+RemoveFiles.prototype.build = function() {
+  for (var i = 0; i < this.inputPaths.length; i++) {
```

Comment:
same as above, persistent filter variant is likely the best bet. This one is likely quite easy to implement that way https://github.com/stefanpenner/broccoli-persistent-filter/blob/master/index.js#L62-L91 should be a reasonable good guide.

The idea is to only apply the require patch, to the last output dir. Rather then having to reconstruct the whole thing.